pkg/util/kernel/download_headers_test.go

```diff
@@ -0,0 +1,213 @@
+}
+
+var debianRepos = []string{
+	"deb http://http.us.debian.org/debian bullseye main",
+	"deb-src http://http.us.debian.org/debian bullseye main",
+	"deb http://security.debian.org/debian-security bullseye-security main",
+	"deb-src http://security.debian.org/debian-security bullseye-security main",
+	"deb http://http.us.debian.org/debian bullseye-updates main",
+	"deb-src http://http.us.debian.org/debian bullseye-updates main",
+}
+
+var targetUbuntu = types.Target{
+	Distro: types.Distro{
+		"ubuntu",
+		"bullseye/sid",
+		"debian",
+	},
+	OSRelease: ubuntuRelease,
+	Uname: types.Utsname{
+		"5.4.0-92-generic",
+		"aarch64",
+	},
+}
+
+var targetDebian types.Target = types.Target{
+	Distro: types.Distro{
+		"debian",
+		"11.2",
+		"debian",
+	},
+	OSRelease: debianRelease,
+	Uname: types.Utsname{
+		"5.10.0-10-arm64",
+		"aarch64",
+	},
+}
+
+type TargetSetup struct {
+	target types.Target
+	repos  []string
+}
+
+var targets = map[string]TargetSetup{
+	"ubuntu": {
+		targetUbuntu,
+		ubuntuRepos,
+	},
+	"debian": {
+		targetDebian,
+		debianRepos,
+	},
+}
+
+func genRepoSuffix(target types.Target) string {
+	return target.Distro.Display + "-" + target.Uname.Kernel + "-" + target.Uname.Machine
+}
+
+func mkTargetDirName(target types.Target) string {
+	return fmt.Sprintf(reposDir, genRepoSuffix(target))
+}
+
+func setup(target types.Target, repos []string, dname string) error {
+	// Make directory where apt config is placed
+	if err := os.MkdirAll(dname, 0744); err != nil {
+		return fmt.Errorf("failed to create dir %s: %w", dname, err)
+	}
+
+	// Make source-list.d
+	sources := fmt.Sprintf(reposSourceDir, genRepoSuffix(target))
+	if err := os.MkdirAll(sources, 0744); err != nil {
+		return fmt.Errorf("failed to create dir %s: %w", sources, err)
+	}
+
+	// add repo source list
+	fname := fmt.Sprintf(reposSource, dname)
+	reposF, err := os.Create(fname)
+	if err != nil {
+		return fmt.Errorf("failed to create file %s: %w", reposSource, err)
+	}
+
+	defer reposF.Close()
+
+	for _, repo := range repos {
+		_, err := reposF.WriteString(repo + "\n")
+		if err != nil {
+			return fmt.Errorf("failed to write repos to file %s: %w", fname, err)
+		}
+	}
+
+	return nil
+}
+
+func getBackend(target *types.Target, reposDir string) (backend types.Backend, err error) {
+	logger := customLogger{}
+	switch strings.ToLower(target.Distro.Display) {
+	case "debian", "ubuntu":
+		backend, err = apt.NewBackend(target, reposDir, logger)
+	default:
+		err = fmt.Errorf("unsupported distribution '%s'", target.Distro.Display)
+	}
+
+	return
+}
+
+func benchmarkHeaderDownload(ts TargetSetup, b *testing.B) {
+	dname := mkTargetDirName(ts.target)
+	kname := fmt.Sprintf(headerDownloadDir, dname)
+
+	if err := setup(ts.target, ts.repos, dname); err != nil {
+		b.Errorf("Failed to setup target: %s\n", err)
+	}
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		// create output directory for kernel headers
+		if err := os.MkdirAll(kname, 0744); err != nil {
```

Comment:
I think we should try to use `t.TempDir()` if we can: https://pkg.go.dev/testing@go1.18.3#T.TempDir